core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -310,6 +337,7 @@ protected void closeInternal() {
             store.decRef();
             indexShard.recoveryStats().decCurrentAsTarget();
         }
+        closedLatch.countDown();
```

Comment:
I was undecided about that one. I wondered if we should only consider successful closing as proper closing for the latch. Looking at it again: As we catch all the exceptions anyhow (IndexOutput.close() is wrapped by `catch (Exception)` and the store is deleted with `deleteQuiet`), there might not be a difference between the two. I'll move it to the finally